src/pages/index.js

```diff
@@ -147,50 +132,73 @@ updateAvatarButton.addEventListener("click", () => {
 });
 
 // ! ||--------------------------------------------------------------------------------||
-// ! ||                                 Functions                                      ||
+// ! ||                                 Functions                                 ||
 // ! ||--------------------------------------------------------------------------------||
 function handleValidation(form) {
   form.enableValidation();
 }
 
 function handleAddCardFormSubmit(cardData) {
-  function makeRequest() {
-    return api.addNewCard(cardData.name, cardData.link).then((res) => {
+  newCardPopup.showButtonProgress(true);
+  api
+    .addNewCard(cardData.name, cardData.link)
+    .then((res) => {
       const card = renderCard(res);
       cardSection.addItem(card);
+      newCardPopup.reset();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      newCardPopup.showButtonProgress(false);
     });
-  }
-  handleSubmit(makeRequest, newCardPopup);
 }
 
 function handleImageClick(name, link) {
   newImagePopup.open(name, link);
 }
 
 function handleProfileSubmit(userData) {
-  function makeRequest() {
-    return api.updateUserInfo(userData.name, userData.about).then((user) => {
+  updateProfilePopup.showButtonProgress(true);
+  api
+    .updateUserInfo(userData.name, userData.about)
+    .then((user) => {
       userInfo.setUserInfo(user);
+      updateProfilePopup.reset();
+      updateProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      updateProfilePopup.showButtonProgress(false);
     });
-  }
-  handleSubmit(makeRequest, updateProfilePopup);
 }
 
 function handleDeleteCard(cardData) {
   deletePopup.open();
   deletePopup.setSubmitAction(() => {
-    function makeRequest() {
-      return api.deleteCard(cardData._id).then(() => {
+    deletePopup.showButtonProgress(true);
+    api
+      .deleteCard(cardData._id)
+      .then((res) => {
         cardData.handleRemoveCard();
+        deletePopup.close();
+      })
+      .catch((err) => {
+        console.error(`Error: ${err}`);
+      })
+      .finally(() => {
+        deletePopup.showButtonProgress(false);
       });
-    }
-    handleSubmit(makeRequest, deletePopup);
   });
 }
 
 function handleLikeIcon(cardData) {
   api
-    .setLike(cardData.id, cardData.isLiked)
+    .setLike(cardData._id, cardData._isLiked)
```

Comment:
* You haven't corrected https://github.com/Marcus3278/se_project_aroundtheus-main/pull/5#discussion_r1612383020  Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.

It should be


```js
    .setLike(cardData.id, cardData.isLiked)


```